routers/web/repo/compare.go

```diff
@@ -567,6 +567,15 @@ func PrepareCompareDiff(
 	} else {
 		title = headBranch
 	}
+	if len(title) > 255 {
+		if ctx.Data["content"] != nil {
+			ctx.Data["content"] = fmt.Sprintf("…%s\n\n%s", title[254:], ctx.Data["content"])
```

Comment:
This is still not correct because you have to consider utf8. Simply use `title[254:]` will break some of utf8 string.

Reply:
Yup looks like we'd need to cast to []rune and then drop the last rune.

Probably also need to consider if the last rune is a combining character too.

Reply:
and we would need to consider this in chi-binding too.

Reply:
I've switched the code to use byte length and truncation algorithm